src/dag-node/addLink.js

```diff
@@ -8,40 +8,30 @@ const DAGLink = require('../dag-link')
 const DAGNode = require('./index')
 const create = require('./create')
 
-function asDAGLink (link, callback) {
+const asDAGLink = async (link) => {
```

Comment:
This is totally a matter of preference but I'd have just add the `async` to the existing function. I tend to reserve arrow functions for when I need an anonymous function and/or closure. I'm not thrilled about the disappearance of `function` across many code bases, I believe it reduces clarity a little bit for the sake of a few characters and something that feels new. Up to you of course!

Reply:
`async` can be removed from this function since `await` isn't used, the call to `toDagLink` will return a `Promise` which will be returned and treated appropriately by the calling function which uses `await` to handle it (or the non-`Promise` returns)

Reply:
Even if the `async` is not needed I like it as an indicator for "this thing returns a Promise". If I remove that, the caller has no indication that a Promise is returned. Am I missing something?

Reply:
Sorry I missed that this function isn't async at all.

Reply:
I still don't get it. I've created this minimal example, that should resemble the calls mentioned here. If I don't have an `async` at `asDagLink()` it will still return a promise:

```js
class Something {
  constructor (value) {
    this.value = value
  }
}

const cid = async () => {
  return 'test'
}

const toDAGLink = async () => {
  const result = await cid()
  return new Something(result)
}

const asDAGLink = () => {
  return toDAGLink()
}

const output = asDAGLink()
console.log(output) // Promise { <pending> }
```

Reply:
I think in this case it comes down to a matter of taste. I can't say for sure that adding `async` will add another `Promise` on top of the one that's returned, I'd hope that VMs optimise that away.
I'm personally happy leaving off `async` from a function when it doesn't contain an `await` and letting the documentation do the work of saying that it returns a `Promise`. Unclear return types are one of the many costs of an untyped language (and `async` only tells you that you can `await` it, not what you're `await`ing, so even this is limited), along with argument types, and we're just used to having to look at metadata (documentation usually) to tell us what an API requires and will return.
And again, `@async` and `@returns Promise` in JSDoc are good tools and can live with your code.

Reply:
Actually ... this function also does a bare return in some cases `return new DAGLink...`, which is fine if you're using `await` but might cause problems for users who expect a `Promise`. So yeah, consistency imposed by `async` is probably best. It'd be like if your example was:

```js
const asDAGLink = () => {
  return Math.random() > 0.5 ? false : toDAGLink()
}
```

Which would be a mess of an API.